lib/services/role.go

```diff
@@ -2738,18 +2743,26 @@ func (set RoleSet) HostUsers(s types.Server) (*HostUsersInfo, error) {
 		for _, group := range role.GetHostGroups(types.Deny) {
 			delete(groups, group)
 		}
-		for _, sudoer := range role.GetHostSudoers(types.Deny) {
-			if sudoer == "*" {
-				sudoers = nil
-				break
+
+		var finalSudoers []string
+	outer:
+		for _, sudoer := range sudoers {
+			for _, deniedSudoer := range role.GetHostSudoers(types.Deny) {
+				if deniedSudoer == "*" {
+					finalSudoers = nil
+					break outer
+				}
+				if sudoer != deniedSudoer {
+					finalSudoers = append(finalSudoers, sudoer)
+				}
```

Comment:
I think that sill this logic is buggy. The final result of `finalSudoers`  will contain a duplication if a sudoers entry is duplicated across different roles 

See: https://github.com/gravitational/teleport/pull/23760/files#diff-1d9446cc64e6fb429df775e3d773435dce3c1e02b49cb0a03ed43ee1d4519f9aR6095